### PR TITLE
Show modal warning users when document fetch results in some errors

### LINKED ISSUE #160: Add modal warning users when they "Download Anyway" if a Download has errored documents.
broken off from #51

### PR DESCRIPTION
**Pull request by amprokop**
Shows modal for the error case when some documents couldn't be fetched.

Fixes #160.

final_efolder_modal

@lakohl — added this GIF to give you an idea of what it currently looks like, and to obtain real content. wanna get that to me when you can?

Test plan: fake a download error 

```
<div class="cf-app-segment cf-app-segment--alt">
  <% if @download.complete? %>
    <% if true %>
```

tab through, check that focus is trapped
check that modal info is marked by aria-labelledby and aria-describedby
check that esc key closes modal
check that focus returns to the previously focused element when modal is closed

cc @KHarshawat

### CODE REVIEW

**Comment by amprokop** [comment]
@shanear — waiting on content from @lakohl @abbyraskin on this, but should be ready for review. thanks!

**Comment by amprokop** [comment]
@shanear — just noticed that CI fails a check. weird bc it passes on my machine. i'll take a look, but the application code itself should still be ready for review.

**Comment by shanear** [comment]
@amprokop my gut tells me it could have to do with a version discrepancy with PhantomJS

File: spec/features/download_spec.rb [hunk 1]
```diff
@@ -293,6 +293,36 @@
...
+    page.execute_script("window.Modal.bind();")
+    expect(page).to have_content "Download anyway"
+    expect(page).to have_no_content "Download incomplete eFolder?"
+
+    within first(".usa-alert-body") do
+      click_on "Download anyway"
+    end
+
+    # check that focus is correctly moved to the modal
+    is_focus_on_modal = page.evaluate_script("document.activeElement.className").include? "cf-modal-startfocus"
```
Comment (amprokop) [inline spec/features/download_spec.rb]: @joofsh — this is the test i was considering removing

File: spec/features/download_spec.rb [hunk 2]
```diff
@@ -293,6 +293,36 @@
...
+    expect(is_focus_on_modal).to be true
+    expect(page).to have_content "Download incomplete eFolder?"
+
+    click_on "Go back"
+
+    # check that focus is correctly moved back to the "Download Anyway" button
+    # after the modal is closed
+    is_focus_on_button = page.evaluate_script("document.activeElement.className").include? "cf-action-openmodal"
```
Comment (amprokop) [inline spec/features/download_spec.rb]: @joofsh @shanear — i also propose to remove this test

**Comment by amprokop** [comment]
interesting. I updated PhantomJS and removed the tests relating to focused elements — and the Travis CI build still fails. It looks like the modal isn't popping up in the phantomJS browser maybe. More to investigate.

**Comment by lakohl** [comment]
@amprokop here copy for the modal:
We encountered errors when retrieving some documents and they won’t be included in the eFolder download. If you elect to 'Download anyway', you may want to retrieve these files individually from VBMS.

**Comment by amprokop** [comment]
@lakohl — one more question too. The content above ^^ uses single quotes. I assume we want double quotes for this, right?

File: app/assets/javascripts/application/modal.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    var escKey = (e.which === 27);
+    var tabKey = (e.which === 9);
+    var tabShift = (e.shiftKey && e.keyCode == 9);
+
+    if (escKey) {
+      $('.cf-modal').trigger('click');
+    }
+
+    if (tabKey) {
+      if ($('.cf-modal-endfocus').is(':focus')) {
```
Comment (amprokop) [inline app/assets/javascripts/application/modal.js]: ahh this actually isn't scoped, you're right

**Comment by amprokop** [comment]
@shanear — i made some changes to scope this more tightly. going to update from master, retest, then merge since you approved this.

**Comment by amprokop** [comment]
image

**Pull request merged**